packages/db/test/unit/column-queries.test.js

```diff
@@ -105,6 +106,76 @@ describe('column queries', () => {
 			expect(queries).to.deep.equal([]);
 		});
 
+		it('should return warning if column type change introduces data loss', async () => {
+			const blogInitial = tableSchema.parse({
+				...userInitial,
+				columns: {
+					date: column.text(),
+				},
+			});
+			const blogFinal = tableSchema.parse({
+				...userInitial,
+				columns: {
+					date: column.date(),
+				},
+			});
+			const { queries, confirmations } = await userChangeQueries(blogInitial, blogFinal);
+			expect(queries).to.deep.equal([
+				'DROP TABLE "Users"',
+				'CREATE TABLE "Users" (_id INTEGER PRIMARY KEY, "date" text NOT NULL)',
+			]);
+			expect(confirmations).to.deep.equal([
+				`Updating existing column ${color.bold(
+					`${TABLE_NAME}.date`
+				)} to a new type that cannot be handled automatically.`,
+			]);
+		});
+
+		it('should return warning if new required column added', async () => {
+			const blogInitial = tableSchema.parse({
+				...userInitial,
+				columns: {},
+			});
+			const blogFinal = tableSchema.parse({
+				...userInitial,
+				columns: {
+					date: column.date({ optional: false }),
+				},
+			});
+			const { queries, confirmations } = await userChangeQueries(blogInitial, blogFinal);
+			expect(queries).to.deep.equal([
+				'DROP TABLE "Users"',
+				'CREATE TABLE "Users" (_id INTEGER PRIMARY KEY, "date" text NOT NULL)',
+			]);
+			expect(confirmations).to.deep.equal([
+				`You added new required column '${color.bold(
+					`${TABLE_NAME}.date`
+				)}' with no default value.` + '\n      This cannot be executed on an existing table.',
+			]);
```

Comment:
We probably shouldn't test this. The messages may be updated at any time and updating the tests could be an inconvenience.

Reply:
in eslint, we would have message IDs rather than asserting on the message itself

maybe we can do similar here, or just assert `confirmations.length` is `1`?

Reply:
Asserting the count sound perfect!